src/shared/components/Tabs/Tabs.tsx

```diff
@@ -1,29 +1,112 @@
+import { BackgroundGradient, Tab, TabsGroup, TabsWrapper, TAB_WIDTH } from './Tabs.styles'
 
-type TabsProps = {
+export type TabsProps = {
   tabs: string[]
   initialIndex?: number
   onSelectTab: (idx: number) => void
 }
 const Tabs: React.FC<TabsProps> = ({ tabs, onSelectTab, initialIndex = -1 }) => {
   const [selected, setSelected] = useState(initialIndex)
+  const [center, setCenter] = useState(0)
+  const tabsRef = useRef<HTMLDivElement>(null)
+  const [shadowsVisible, setShadowsVisible] = useState({
+    left: false,
+    right: true,
+  })
+
+  useResizeObserver<HTMLDivElement>({
+    ref: tabsRef,
+    onResize: ({ width }) => {
+      if (!tabsRef.current || !width) {
+        return
+      }
+      const isScrollBarVisible = tabsRef.current.scrollWidth > width
+      if (isScrollBarVisible) {
+        setCenter(width / 3)
+      } else {
+        setCenter(0)
+      }
+    },
+  })
+
+  useEffect(() => {
+    const tabsGroup = tabsRef.current
+    if (!tabsGroup) {
+      return
+    }
+    const currentItemOffsetleft = selected > 0 && TAB_WIDTH * selected
+    if (currentItemOffsetleft) {
+      tabsGroup.scrollLeft = currentItemOffsetleft - center
+    }
+  }, [center, selected, tabs.length, tabsRef])
+
+  useEffect(() => {
+    const tabsGroup = tabsRef.current
+    if (!tabsGroup) {
+      return
+    }
+    const touchHandler = throttle(() => {
+      const { scrollLeft, clientWidth, scrollWidth } = tabsGroup
+      if (scrollLeft > 10) {
+        setShadowsVisible((shadows) => ({ ...shadows, left: true }))
+      } else {
+        setShadowsVisible((shadows) => ({ ...shadows, left: false }))
+      }
+      if (scrollLeft < scrollWidth - clientWidth - 10) {
+        setShadowsVisible((shadows) => ({ ...shadows, right: true }))
+      } else {
+        setShadowsVisible((shadows) => ({ ...shadows, right: false }))
+      }
+    }, 100)
+    tabsGroup.addEventListener('touchmove', touchHandler)
+    tabsGroup.addEventListener('scroll', touchHandler)
+    return () => {
+      touchHandler.cancel()
+      tabsGroup.removeEventListener('touchmove', touchHandler)
+      tabsGroup.removeEventListener('scroll', touchHandler)
+    }
+  }, [])
+
+  const isGradientVisible = center > 0
 
   return (
-    <TabsGroup>
-      {tabs.map((tab, idx) => (
-        <Tab
-          onClick={(e) => {
-            onSelectTab(idx)
-            setSelected(idx)
-          }}
-          key={`${tab}-${idx}`}
-          selected={selected === idx}
-        >
-          <span>{tab}</span>
-        </Tab>
-      ))}
-    </TabsGroup>
+    <TabsWrapper>
+      <CSSTransition
+        in={shadowsVisible.left && isGradientVisible}
+        timeout={100}
+        classNames={transitions.names.fade}
+        unmountOnExit
+      >
+        <BackgroundGradient direction="prev" />
+      </CSSTransition>
+      <TabsGroup ref={tabsRef}>
+        {tabs.map((tab, idx) => (
+          <Tab
+            onClick={() => {
```

Comment:
I know this is carried from the old code, but maybe we could elevate that function to a named handler to make it more readable? We can use the `createClickHandler` pattern